### PR TITLE
Fixes #24805 - Clean up remaining taxonomy enabled checks

### PR DESCRIPTION
**Pull request by tbrisker**
Also removes related methods that have been deprecated since 1.21.

### CODE REVIEW

File: app/models/concerns/taxonomix.rb [hunk 1]
```diff
@@ -195,9 +195,7 @@ def children_of_selected_organization_ids
   # for existing records it would block saving global objects if taxonomies were not changed
   # for new records this would allow to create global objects for non global users
   def ensure_taxonomies_not_escalated
-    taxonomies = []
-    taxonomies << Organization if SETTINGS[:organizations_enabled]
-    taxonomies << Location if SETTINGS[:locations_enabled]
+    taxonomies = [Organization, Location]
```
Comment (amirfefer) [inline app/models/concerns/taxonomix.rb]: nitpick: could you move it to a helper? there are few identical usages.

**Review by amirfefer — COMMENTED** [review]
Thanks @tbrisker :+1: ,
 I left an inline comment.

File: app/models/concerns/taxonomix.rb (hunk 1 above)
Comment (tbrisker) [reply app/models/concerns/taxonomix.rb]: Would this be easier to understand?

```ruby
    taxonomies = Taxonomy.subclasses
```

Comment (timogoebel) [reply app/models/concerns/taxonomix.rb]: I'd prefer a dedicated method. Don't we have `Taxonomy.all_enabled_taxonomies` or something? I get, that the enabled doesn't make sense anymore. But something similar would be good. `Taxonomy.all_kinds` or `Taxonomy.all_available` or ...

Comment (timogoebel) [reply app/models/concerns/taxonomix.rb]: Ah, should have read the whole PR. I do like `enabled_taxonomies`, just the naming needs an update imho.

Comment (tbrisker) [reply app/models/concerns/taxonomix.rb]: How about `Taxonomy.types`?

Comment (timogoebel) [reply app/models/concerns/taxonomix.rb]: Sure, why not.

Comment (tbrisker) [reply app/models/concerns/taxonomix.rb]: added and changed other places that had the same array to use it.

**Review by timogoebel — APPROVED** [review]

**Review by ekohl — APPROVED** [review]
The `Taxonomy.types` method looks like something we should have implemented even when they were configurable.

**Comment by mmoll** [comment]
[test foreman]

**Comment by tbrisker** [comment]
All :green_apple:

**Comment by mmoll** [comment]
merged, thanks @tbrisker!

**Pull request merged**